test/parallel/sdk.API.claimRewards.test.ts

```diff
@@ -0,0 +1,450 @@
+/*
+ *
+ * @group sdk
+ */
+import { jest } from "@jest/globals";
+import { Keyring } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { BN } from "@polkadot/util";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { User } from "../../utils/User";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { Xyk } from "../../utils/xyk";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import {
+  claimRewardsAll,
+  getLiquidityAssetId,
+  getRewardsInfo,
+  promotePool,
+} from "../../utils/tx";
+import {
+  getEventResultFromMangataTx,
+  getNextAssetId,
+} from "../../utils/txHandler";
+import { ExtrinsicResult, waitForRewards } from "../../utils/eventListeners";
+import { BN_ZERO } from "@mangata-finance/sdk";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser: User;
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let soloTokenId: BN;
+let poolTokenIds: BN[];
+let liqIds: BN[];
+let tokenAmounts: BN[];
+const defaultCurrencyValue = new BN(250000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  await setupApi();
+  await setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintNative(sudo)
+  );
+
+  [testUser] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+
+  [soloTokenId] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue],
+    sudo
+  );
+
+  const batchPromisesMinting = [];
+  const batchPromisesPromoting = [];
+  poolTokenIds = [];
+  tokenAmounts = [];
+  liqIds = [];
+
+  const nextTokenId = (await getNextAssetId()).toNumber();
+
+  for (
+    let newTokenId = nextTokenId;
+    newTokenId < nextTokenId + 13;
```

Comment:
+13? perhaps a comment about what are you trying to do on this loop would help :)

Reply:
We need to create 13 pools - 12 for checking that if a user has available some rewards in over ten pools then we receive error and one for checking deactivated pool

Reply:
gotcha! thanks!